x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,297 @@
+ */
+export function getEnterpriseSearchNodeDeprecation(config: ConfigType, isCloud: boolean): DeprecationsDetails[] {
+  if (config.host) {
+    const steps = []
+    if (isCloud) {
+      steps.push(...[
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.gotocloud', {
+          defaultMessage: 'Go to cloud.elastic.co',
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clickedit', {
+          defaultMessage: "Click the 'Edit' tab",
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.scrolldown', {
+          defaultMessage: "Scroll down to the 'Enterprise Search' section",
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clickx', {
+          defaultMessage: "Click the red 'X' by 'Enterprise Search instances'",
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.nocapacity', {
+          defaultMessage: 'You should no longer see any Enterprise Search capacity',
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clicksave', {
+          defaultMessage: "Click 'Save' and confirm",
+        }),
+      ])
+    } else {
+      steps.push(...[
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.stopent', {
+          defaultMessage: 'Stop all your Enterprise Search nodes',
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.stopkibana', {
+          defaultMessage: 'Stop all your Kibana nodes',
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.removeconfig', {
+          defaultMessage: "Edit 'kibana.yml' to remove 'enterpriseSearch.host'",
+        }),
+        i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.restart', {
+          defaultMessage: "Restart Kibana",
+        }),
+      ])
+    }
+    return [{
+      level: 'critical',
+      deprecationType: 'feature',
+      title: i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.title', {
+        defaultMessage: 'Enterprise Search host(s) must be removed',
+      }),
+      message: i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.message', {
+        defaultMessage: 'Enterprise Search is not supported in versions >= 9.x. ' +
+          'You must remove Enterprise Search node(s) from your deployment to proceed with the upgrade. ' +
+          'Note that once Enterprise Search is stopped, products such as App Search, Workplace Search, and Elastic Crawler ' +
+          'will cease to operate. Native Connectors will also stop running syncs, between when Enterprise Search is stopped ' +
+          'and when the 9.x upgrade completes. For full details, see the documentation.',
+      }),
+      documentationUrl: 'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
```

Comment:
This docs link 404's right now, but I've been told is the home where upgrade guidance _will_ be. It does exist on the `main` branch

Reply:
This is the correct URL, but not in `current` - it works with `master`

Reply:
Right - I think we probably don't want to merge it with a link to `master` though, and want to assume that by the time customers see this in action, `current` will be correct. WDYT?

Reply:
`current` will be correct. yes.

Reply:
@sphilipse @efegurkan @markjhoy don't we need to create a doclink, rather than hardcoding current here?

Also, there won't be a 9.x branch on the Enterprise Search guide, so current on Kibana won't line up with current on the Enterprise Search guide when 9.0.0 releases, so I guess if we were to hardcode anything, it would be `8.last`?

Reply:
I was able to hook in a docslink. And you're right, it'll never be `current`, but will be `8.18` and `8.19`, but that's what we'll want, I believe.